From 173ad621b7a74e57f3f76c1fb944f238ae85bf97 Mon Sep 17 00:00:00 2001
From: Jammy Huang <jammy_huang@aspeedtech.com>
Date: Wed, 16 Feb 2022 15:29:30 +0800
Subject: [PATCH] Modify Content-Security-Policy(CSP) to adapt WebAssembly

CompileError, "Wasm code generation disallowed by embedder",
happens for WebAssembly.

Add 'unsafe-eval' to script-src of CSP to fix it.

Signed-off-by: Jammy Huang <jammy_huang@aspeedtech.com>
---
 include/security_headers.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/security_headers.hpp b/include/security_headers.hpp
index 2a2eb40d..1228d32f 100644
--- a/include/security_headers.hpp
+++ b/include/security_headers.hpp
@@ -65,7 +65,7 @@ inline void addSecurityHeaders(const crow::Request& req [[maybe_unused]],
                                                  "img-src 'self' data:; "
                                                  "font-src 'self'; "
                                                  "style-src 'self'; "
-                                                 "script-src 'self'; "
+                                                 "script-src 'self' 'unsafe-eval'; "
                                                  "connect-src 'self' wss:; "
                                                  "form-action 'none'; "
                                                  "frame-ancestors 'none'; "
